app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Services\UserService;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+use App\Traits\APIResponse;
+
+class UserController extends ControllerWithGuard
+{
+    use APIResponse;
+
+    private $userService;
+
+    public function __construct(UserService $userService)
+    {
+        parent::__construct(); 
+        $this->userService = $userService;
+    }
+
+    public function index()
+    {
+        $result = $this->userService->index();
+        return $this->responseSuccessWithData($result);
+    }
+
+    public function store(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+            'name' => 'required|string|max:255',
+            'email' => 'required|string|email|unique:users',
+            'password' => 'required|string|min:6',
+        ]);
+
+        if ($validator->fails()) {
+            return response()->json([
+                'success' => false,
+                'errors' => $validator->errors()
+            ], 422);
+        }
+
+        $result = $this->userService->store($request->name, $request->email, $request->password);
+
+        return $this->responseSuccessWithData($result, true);
+    }
+
+    public function show($id)
+    {
+        $user = $this->userService->show($id);
+
+        if (!$user) {
+            return response()->json(['message' => 'User not found'], 404);
+        }
+
+        return $this->responseSuccessWithData($user);
+    }
+
+    public function update($id, Request $request)
+    {
+        $result = $this->userService->update($id, ['name'=>($request->name), 'email' => ($request->email), 'password'=> ($request->password)]);
```

Comment:
The array passed to the update method is not formatted consistently. Consider formatting it for better readability:
```suggestion
        $result = $this->userService->update($id, [
            'name' => $request->name,
            'email' => $request->email,
            'password' => $request->password,
        ]);
```